hw2.sql

```diff
@@ -0,0 +1,161 @@
+--Formatted with https://www.dpriver.com/pp/sqlformat.htm
+
+USE baseball;
+
+--Create hits table with only hits and no outs
+CREATE TABLE only_hits AS
+  SELECT *
+  FROM   hits
+  WHERE  type = 'H';
+
+--Append local date to hits table so i dont have to join with game table every time
+CREATE TABLE hits_with_date AS
+  SELECT h.*,
+         g.local_date
+  FROM   only_hits h
+         JOIN game g
+           ON h.game_id = g.game_id;
+
+--append local date to atbats table
+CREATE TABLE atbats_with_date AS
+  SELECT ab.*,
+         g.local_date
+  FROM   atbat_r ab
+         JOIN game g
+           ON ab.game_id = g.game_id;
+
+--create hits by year table
+CREATE TABLE hits_by_year AS
+  SELECT Count(hits_id)   AS hits,
+         batter,
+         Year(local_date) AS year
+  FROM   hits_with_date
+  GROUP  BY batter,
+            year;
+
+--create atbats by year table
+CREATE TABLE atbats_by_year AS
+  SELECT Count(atbat_r_id) AS atbats,
+         batter,
+         Year(local_date)  AS year
+  FROM   atbats_with_date
+  GROUP  BY batter,
+            year;
+
+--create batting averge by year table
+CREATE TABLE batting_avg_by_year AS
+  SELECT hits / atbats AS batting_average,
+         hits,
+         atbats,
+         h.batter,
+         h.year
+  FROM   hits_by_year h
+         JOIN atbats_by_year ab
+           ON h.batter = ab.batter
+              AND h.year = ab.year;
+
+--create historic hits utilizing hits_by_year
+CREATE TABLE historic_hits AS
+  SELECT Sum(hits) AS hits,
+         batter
+  FROM   hits_by_year
+  GROUP  BY batter;
+
+--create historic atbats using atbats_by_year
+CREATE TABLE historic_atbats AS
+  SELECT Sum(atbats) AS atbats,
+         batter
+  FROM   atbats_by_year
+  GROUP  BY batter;
+
+--create historic batting average table
+CREATE TABLE historic_batting_average AS
+  SELECT hits / atbats AS batting_average,
+         h.batter
+  FROM   historic_hits h
+         JOIN historic_atbats ab
+           ON h.batter = ab.batter;
+
+--create hits by date table used in the rolling ba
+CREATE TABLE hits_by_date AS
+  SELECT Count(hits_id) AS hits,
+         batter,
+         local_date
+  FROM   hits_with_date
+  GROUP  BY batter,
+            local_date;
+
+--create at bats by date table used in the rolling ba
+CREATE TABLE atbats_by_date AS
+  SELECT Count(atbat_r_id) AS atbats,
+         batter,
+         local_date
+  FROM   atbats_with_date
+  GROUP  BY batter,
+            local_date;
+
+--create rolling count of hits
+CREATE TABLE hits_rolling AS
```

Comment:
This seems to take a very long time for me to execute, but it's running in a few minutes on your end...  We might need to figure out why.